ui/app/src/components/GitAuthForm/GitAuthForm.tsx

```diff
@@ -0,0 +1,259 @@
+                  <IconButton edge="end" aria-label="toggle password visibility" onClick={handleClickShowPassword}>
+                    {showPassword ? <VisibilityOff /> : <Visibility />}
+                  </IconButton>
+                </InputAdornment>
+              )
+            }}
+          />
+        )}
+        {type === 'token' && (
+          <TextField
+            id="repoToken"
+            name="repoToken"
+            className={clsx(classes.margin, classes.textField)}
+            type={showPassword ? 'text' : 'password'}
+            label={formatMessage(messages.token)}
+            required
+            value={inputs.repoToken}
+            error={inputs.submitted && !inputs.repoToken}
+            helperText={renderHelperText(formatMessage(messages.token), inputs.repoToken, '', true, inputs.submitted)}
+            onKeyPress={onKeyPress}
+            onChange={handleInputChange}
+            InputProps={{
+              endAdornment: (
+                <InputAdornment position="end">
+                  <IconButton edge="end" aria-label="toggle password visibility" onClick={handleClickShowPassword}>
+                    {showPassword ? <VisibilityOff /> : <Visibility />}
+                  </IconButton>
+                </InputAdornment>
+              )
+            }}
+          />
+        )}
+        {type === 'key' && (
+          <TextField
+            id="repoKey"
+            name="repoKey"
+            label={formatMessage(messages.privateKey)}
+            required
+            fullWidth
+            multiline
+            className={classes.margin}
+            error={inputs.submitted && !inputs.repoKey}
+            helperText={renderHelperText(
+              formatMessage(messages.privateKey),
+              inputs.repoKey,
+              '',
+              true,
+              inputs.submitted
+            )}
+            onKeyPress={onKeyPress}
+            onChange={handleInputChange}
+            value={inputs.repoKey}
+          />
+        )}
+      </div>
+    );
+  }
+
+  return (
+    <RadioGroup
+      aria-label="repoAuthentication"
+      name="repoAuthentication"
+      value={inputs.repoAuthentication}
+      onChange={handleInputChange}
+      onKeyPress={onKeyPress}
+    >
+      <FormControlLabel
+        value="none"
+        control={<Radio onChange={() => viewAuth('none')} />}
+        label={
+          <FormattedMessage
+            id="gitForm.noAuthenticationRequired"
+            defaultMessage="Authentication not required (public URL)"
+          />
+        }
+      />
+      <FormControlLabel
+        value="basic"
+        control={<Radio onChange={() => viewAuth('basic')} />}
+        label={<FormattedMessage id="gitForm.usernameAndPassword" defaultMessage="Username & Password" />}
+      />
+      <Collapse in={inputs.expanded.basic} timeout={300} unmountOnExit>
+        {inputs.expanded.basic && renderAuth(inputs.repoAuthentication)}
+      </Collapse>
```

Comment:
There's some duplication of logic in **all these**, isn't there? If `umountOnExit`, the additional internal conditional is not necessary?